### ISSUE TITLE
Evaluation/Inference after training

### ISSUE THREAD

**Issue by suchiek** [fizyr/keras-retinanet#420]
Hi,

Hope this is not too long winded. I have a couple of questions here regarding evaluation and inference.

Part 1:
I trained a model on my own data and the classification loss fell steadily from .98 to .21 and on running  evaluation, I got the following results for the categories that I am trying to predict:

UprightLamp 0.0103
MetalPole 0.5370
WoodPole 0.6929
PowerHead 0.4279
Attachment 0.4786
TrafficSignal 0.4813
mAP: 0.4380

Given the downward trend in classification loss during training are these values of precision to be expected? 

Part 2:
I converted the training model to an inference model and used the Jupyter notebook on your site to get bounding boxes and predictions for an image. Here is the code that I used:
```python
# show images inline
%matplotlib inline

# automatically reload modules when they have changed
%load_ext autoreload
%autoreload 2

# import keras
import keras

# import keras_retinanet
from keras_retinanet.models.resnet import custom_objects
from keras_retinanet.utils.image import read_image_bgr, preprocess_image, resize_image
from keras_retinanet.utils.visualization import draw_box, draw_caption
from keras_retinanet.utils.colors import label_color

# import miscellaneous modules
import matplotlib.pyplot as plt
import cv2
import os
import numpy as np
import time

import tensorflow as tf

def get_session():
    config = tf.ConfigProto()
    config.gpu_options.allow_growth = True
    return tf.Session(config=config)

# set the modified tf session as backend in keras
keras.backend.tensorflow_backend.set_session(get_session())

#model that I trained and then converted to an inference model:
model_path = "/home/mdsskv/keras-retinanet/keras_retinanet/resnet50_csv_12.h5"

model = load_model(model_path, backbone='resnet50')
print(model.summary())

# load label to names mapping for visualization purposes
labels_to_names ={0: 'UprightLamp', 1: 'MetalPole', 2: 'WoodPole', 3: 'PowerHead', 4: 'Attachment', 5: 'TrafficSignal'}
```
At the end of this section, I got the following message:

UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
warnings.warn('No training configuration found in save file: 

The image yielded no predictions.

thanks for your input!

**Comment by suchiek** [comment]
Thank you so much for your response and this great repository.
